server/internal/infrastructure/aws/file.go

```diff
@@ -0,0 +1,276 @@
+package aws
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/google/uuid"
+	"github.com/reearth/reearth-cms/server/internal/usecase/gateway"
+	"github.com/reearth/reearth-cms/server/pkg/asset"
+	"github.com/reearth/reearth-cms/server/pkg/file"
+	"github.com/reearth/reearthx/i18n"
+	"github.com/reearth/reearthx/rerror"
+	"github.com/samber/lo"
+)
+
+const (
+	s3AssetBasePath string = "assets"
+	fileSizeLimit   int64  = 10 * 1024 * 1024 * 1024 // 10GB
+)
+
+type fileRepo struct {
+	bucketName   string
+	baseURL      *url.URL
+	cacheControl string
+	s3Client     *s3.Client
+}
+
+func NewFile(bucketName, accessKeyID, secretAccessKey, region, baseURL, cacheControl string) (gateway.File, error) {
+	if bucketName == "" {
+		return nil, rerror.NewE(i18n.T("bucket name is empty"))
```

Comment:
```suggestion
		return nil, errors.New("bucket name is empty")
```

This error is not displayed to the user and does not require translation.

Reply:
Done!